### PR TITLE
Port :erlang.round/1 to JS

### LINKED ISSUE #566: Port :erlang.round/1 to JS

### PR DESCRIPTION
**Pull request by bartblast**
Closes #566

## Summary by CodeRabbit

* **New Features**
  * Added `round/1` function to Erlang support for rounding numeric values to the nearest integer with consistent behavior.

* **Tests**
  * Added extensive test coverage including positive and negative numbers, fractional edge cases, safe integer boundaries, zero variations, and validation of error handling for non-numeric inputs.

### CODE REVIEW

File: assets/js/erlang/erlang.mjs [hunk 1]
```diff
@@ -1898,6 +1898,28 @@ const Erlang = {
...
+      return number;
+    }
+
+    // Erlang rounds half away from zero (5.5 -> 6, -5.5 -> -6)
+    // JavaScript Math.round rounds half toward positive infinity (5.5 -> 6, -5.5 -> -5)
+    // Use sign * round(abs) to get correct behavior
+    // Adding 0 converts -0 to 0
+    const value = number.value;
+    return Type.integer(Math.sign(value) * Math.round(Math.abs(value)) + 0);
+  },
```
[the remark this answers is not in the document]
Comment (bartblast) [reply assets/js/erlang/erlang.mjs:1919]: NaN and Infinity values cannot occur in normal Hologram operation since they don't exist in Erlang/Elixir - the BEAM VM doesn't support these values. Additionally, our testing conventions require JS tests to match Elixir consistency tests exactly, so we wouldn't be able to test this guard. Since the code path is unreachable in practice and untestable within our conventions, I opted not to add it.

File: test/javascript/erlang/erlang_test.mjs [hunk 2]
```diff
@@ -6662,6 +6662,147 @@ describe("Erlang", () => {
...
+    });
+
+    it("handles large float that loses precision", () => {
+      // This demonstrates that float representation limits apply before rounding
+      // 36_028_797_018_963_969.0 cannot be represented exactly as a float
+      // It's stored as 36_028_797_018_963_968.0
+      // eslint-disable-next-line no-loss-of-precision
+      const result = testedFun(Type.float(36_028_797_018_963_969.0));
+      const expected = Type.integer(36_028_797_018_963_968n);
+
```
[the remark this answers is not in the document]
Comment (bartblast) [reply test/javascript/erlang/erlang_test.mjs:6793]: Our CI only runs ESLint (`mix eslint`), not Biome. While Biome is listed as a dependency, the `eslint-disable` comments are the appropriate ones for suppressing warnings in our pipeline.
